src/compiler/fs-watch/fs-watch-rebuild.ts

```diff
@@ -8,47 +8,85 @@ export const filesChanged = (buildCtx: d.BuildCtx) => {
   return unique([...buildCtx.filesUpdated, ...buildCtx.filesAdded, ...buildCtx.filesDeleted]).sort();
 };
 
-export const scriptsAdded = (buildCtx: d.BuildCtx) => {
-  // collect all the scripts that were added
-  return buildCtx.filesAdded
-    .filter((f) => {
-      return SCRIPT_EXT.some((ext) => f.endsWith(ext.toLowerCase()));
-    })
-    .map((f) => basename(f));
-};
-
-export const scriptsDeleted = (buildCtx: d.BuildCtx) => {
-  // collect all the scripts that were deleted
-  return buildCtx.filesDeleted
-    .filter((f) => {
-      return SCRIPT_EXT.some((ext) => f.endsWith(ext.toLowerCase()));
-    })
-    .map((f) => basename(f));
-};
-
-export const hasScriptChanges = (buildCtx: d.BuildCtx) => {
-  return buildCtx.filesChanged.some((f) => {
-    const ext = getExt(f);
-    return SCRIPT_EXT.includes(ext);
-  });
-};
-
-export const hasStyleChanges = (buildCtx: d.BuildCtx) => {
-  return buildCtx.filesChanged.some((f) => {
-    const ext = getExt(f);
-    return STYLE_EXT.includes(ext);
-  });
-};
-
-const getExt = (filePath: string) => filePath.split('.').pop().toLowerCase();
-
+/**
+ * Unary helper function mapping string to string and wrapping `basename`,
+ * which normally takes two string arguments. This means it cannot be passed
+ * to `Array.prototype.map`, but this little helper can!
+ * @param filePath a filepath to check out
+ * @returns the basename for that filepath
+ */ 
+const unaryBasename = (filePath: string): string => basename(filePath);
+
+/**
+ * Get the file extension for a path
+ * @param filePath a path
+ * @returns the file extension (well, characters after the last `'.'`)
+ */
+const getExt = (filePath: string): string => filePath.split('.').pop().toLowerCase();
+
+/**
+ * Script extensions which we want to be able to recognize
+ */
 const SCRIPT_EXT = ['ts', 'tsx', 'js', 'jsx'];
-export const isScriptExt = (ext: string) => SCRIPT_EXT.includes(ext);
+
+/**
+ * Helper to check if an extension is found in our list of script extensions
+ * @param ext a file extension
```

Comment:
Should this be something more like:
```suggestion
 * Helper to check if an extension is found in our list of script extensions
 * @param filePath the path of a file to check
```

I'm not entirely sure the summary on the first line aligns with what we're doing here. Thoughts?